File: features/lead/src/main/java/com/nivasafinance/features/leadlender/exception/LeadNotFoundException.java
package com.nivasafinance.features.leadlender.exception;

import com.nivasafinance.common.exception.ResourceNotFoundException;

import java.io.Serial;

public class LeadNotFoundException extends ResourceNotFoundException {
    
    @Serial
    private static final long serialVersionUID = 1L;
    
    public LeadNotFoundException(String message) {
        super(message);
    }
}

File: features/lead/src/main/java/com/nivasafinance/features/lead/lead/exception/LeadNotFoundException.kt
package com.nivasafinance.features.lead.lead.exception

import exception.ResourceNotFoundException
import org.springframework.context.MessageSource
import org.springframework.context.i18n.LocaleContextHolder
import java.util.Locale
import java.util.UUID

/**
 * Exception thrown when a lead is not found by ID or other criteria.
 */
class LeadNotFoundException(message: String) : ResourceNotFoundException(message) {
    companion object {
        private val locale: Locale = LocaleContextHolder.getLocale()
        const val LEAD_ID_NOT_FOUND_KEY = "error.lead.id.not.found"
    }

    /**
     * Constructor with a lead ID.
     * This is a convenience constructor for the common case of a lead not found by ID.
     */
    constructor(id: UUID, messageSource: MessageSource) : this(
        messageSource.getMessage(LEAD_ID_NOT_FOUND_KEY, arrayOf(id), locale)
    )
}
